### PR TITLE
Add hybrid planning tutorial

### PR DESCRIPTION
**Pull request by sjahr**
### Description

A new tutorial for MoveIt 2's hybrid planning feature. WIP and blocked till the feature got merged into the MoveIt2's main branch ([moveit2 #488](https://github.com/ros-planning/moveit2/pull/488))
### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by vatanaksoytezer** [comment]
@sjahr do you mind tagging me when this is ready for review?

**Comment by sjahr** [comment]
@vatanaksoytezer This PR is ready to review. Thanks! The code links won't work because the hybrid planning feature is not merged yet. Not sure if we should wait for it with merging this PR after the review.

File: doc/hybrid_planning/hybrid_planning_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To get a global motion planning solution, the *Global Planner Component* needs to be activated via the *Global Planning Action Server*. When it receives a *MotionPlanRequest* the component computes a motion plan with the *Global Planner Plugin* and publishes the solution to the other components.
+The dataflow within the component can be seen in the picture below:
+
+.. image:: images/global_planner_dataflow.png
+   :width: 500pt
+
+The *Global Planner Plugin* can be used to implement and customize the global planning algorithm. To implement you own planner you simply need to inherit from the :moveit2_codedir:`GlobalPlannerInterface <moveit_ros/hybrid_planning/global_planner/global_planner_component/include/moveit/global_planner/global_planner_interface.h>`: ::
+
+   class MySmartPlanner : public GlobalPlannerInterface
```
Comment (tylerjw) [inline doc/hybrid_planning/hybrid_planning_tutorial.rst:105]: Consider renaming `GlobalPlannerInterface` to `GlobalPlannerPlugin` to make it clear this is the plugin interface for a global planner.

**Review by tylerjw — COMMENTED** [review]
Thank you for writing such a nicely detailed tutorial.  I left a few comments.

File: doc/hybrid_planning/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+set(INCLUDE_DEPENDS
+  rclcpp
+  rclcpp_action
+  moveit_core
+  moveit_ros_planning
+  moveit_msgs
+)
+
+add_executable(hybrid_planning_test_node src/hybrid_planning_test_node.cpp)
```
[the remark this answers is not in the document]
Comment (AndyZe) [reply doc/hybrid_planning/CMakeLists.txt]: I went ahead an made a commit to remove the launch/config/src files from this tutorial. They're practically duplicated in the moveit source code. In fact, I think they're exactly duplicated.

[the remark this answers is not in the document]
Comment (sjahr) [reply doc/hybrid_planning/CMakeLists.txt]: Yes, I am aware of this. My plan was to remove them from the moveit source code, once this PR got merged but it is probably a better idea to add them later after hybrid planning got merged (if they should be moved to the tutorials)

**Comment by vatanaksoytezer** [comment]
@AndyZe can you add ros2 branch of moveit_msgs to [repos file](https://github.com/ros-planning/moveit2_tutorials/blob/main/moveit2_tutorials.repos) so that CI is happy and we can merge this.

**Pull request merged**